common/fs/src/cache/tailed_file.rs

```diff
@@ -388,6 +393,29 @@ impl LineMetaMut for LazyLineSerializer {
     }
 }
 
+impl LineBufferMut for LazyLineSerializer {
+    fn get_line_buffer(&mut self) -> Option<&[u8]> {
+        if self.line_buffer.as_ref().is_some() {
+            // Get the value without locking
+            return self.line_buffer.as_deref();
+        }
+
+        match self.reader.try_lock() {
+            Some(file_inner) => {
+                // Cache the value to avoid further cloning
+                self.line_buffer = Some(Bytes::from(file_inner.buf.clone()));
```

Comment:
This is doing approximately the same thing as https://github.com/logdna/logdna-agent-v2/pull/119/files#diff-a50000c4ee02c6f3895eebe2a870cac29962b2bb769f3cfad31793688e2de3d8R288, except this one is including the last character, which is always a trailing new line. Is that intentional?

Reply:
Nice catch, I'll fix it.